src/util/createElements.js

```diff
@@ -160,6 +160,13 @@ module.exports = async (elements, jobId, processId) => {
               }
             });
           }
+          if (!isNilOrEmpty(element.configuration) && !isNilOrEmpty(existingElement.id)) {
+            element.id = existingElement.id;
+            promisesList.updatedConfiguration.push(
+              update(`elements/${existingElement.id}/configurations`, element)
```

Comment:
we don't wanna update all configurations, For the existing configuration, just call update API and for non-existing configuration, call POST API. Also, why are you sending the entire element as a payload?

Reply:
@MohdRafiUddin We should not make a call config by config one by one, because we should not be able to update the configs (like pagination, page max) which we get for all system catalog elements and we can't differentiate which config is not system config.

The api PUT /configurations is created for this purpose only to extend configs for elements ( Soba PR 👉 https://github.com/cloud-elements/soba/pull/12496 ) and it is written in a way that we can't update the configs like pageSize. This is the same api we are using in UI, when we click save from configurations page. The request body for this api is entire element payload, so I'm passing the same.

I tried uploading the new elements and it is working as expected. This code won't affect the new elements as there is a condition in the same file, if element is private we are using POST /elements.

Reply:
In the case of the extended element, why would you need to update pagination, page max?